### ISSUE TITLE
Active Learning GetLowScoreVariation returns no results if matching score is greater than 0.95

### ISSUE THREAD

**Issue by priests123** [microsoft/botbuilder-dotnet#3642]
## Version
I'm using 4.7.2 but exists in 4.8

## Describe the bug
In the [ActiveLearningUtils.cs](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder.AI.QnA/Utils/ActiveLearningUtils.cs) there is a method called GetLowScoreVariation. I have some search results from QnA being passed in with the following scores:

- Result A score = 1
- Result B score = 0.5087

I would expect this method to return Result A but it is returning no results. Looking at the logic, the if statement on line 56 results in returning false as the topScoreAnswer I have, 1, isn't less than the MaxScoreForLowScoreVariation of 0.95.

## Expected behaviour
I would expect this method to return the highest scoring result if the result is greater than the MaxScoreForLowScoreVariation

**Comment by v-kydela** [comment]
Thank you for the feedback, @priests123

I think I do see some strange behavior in GetLowScoreVariation, but I'm not sure if it's the behavior you're talking about. If you had only one answer and it was greater than MaxScoreForLowScoreVariation, that answer would be returned. Is that why you'd also expect the answer to be returned when there's more than one answer?

You say "I would expect this method to return the highest scoring result if the result is greater than the MaxScoreForLowScoreVariation" but that seems counterintuitive. Why would you expect the method to work with a value that's higher than its stated maximum? Are you saying you don't think there should be a maximum

@trojenguri 
@EricDahlvang 

Based on my analysis, I think this class/method could use some reworking. There seem to be some inconsistencies and redundancies.

**Comment by priests123** [comment]
See my responses:

If you had only one answer and it was greater than MaxScoreForLowScoreVariation, that answer would be returned. Is that why you'd also expect the answer to be returned when there's more than one answer?

**Yes, I would expect an answer to be returned if there were multiple values and one was greater than the MaxScoreForLowScoreVariation. I would think too there is the case that multiple results passed to this could be greater than the MaxScoreForLowScoreVariation**

You say "I would expect this method to return the highest scoring result if the result is greater than the MaxScoreForLowScoreVariation" but that seems counterintuitive. Why would you expect the method to work with a value that's higher than its stated maximum? Are you saying you don't think there should be a maximum?

**I would expect it to work as there can be scores that are greater then the MaxScoreForLowScoreVariation. If there is a single QueryResult passed to it this, it works as expected. The issue is when multiple QueryResults are passed to it and 1 (or more) are greater than the MaxScoreForLowScoreVariation**

In my code, I'm using this method to filter ambiguous questions before I do active learning. Looking at the docs this is the method it has to call.

**Comment by trojenguri** [comment]
Thanks for the feedback. But as a mitigation, before calling LowScoreVariation function, you can have a check 
```
if (isActiveLearningEnabled && response.Answers.Any() && response.Answers.First().Score <= maximumScoreForLowScoreVariation)
            {
                // Get filtered list of the response that support low score variation criteria.
                response.Answers = qnaClient.GetLowScoreVariation(response.Answers);
}
```

You can refer to the QnAMakerDialog on how it is implemented. https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog.cs
Line no. 215.

**Comment by priests123** [comment]
Hi @trojenguri,

I have added similar logic into my code. Would that code not suitable to be included in the method rather then having to always include it when calling it?

**Comment by gurvsing** [comment]
Yes, I agree. That is something that we missed, I will add a task to move this into the function. Thanks

**Comment by vipeketi** [comment]
Fixed with https://github.com/microsoft/botbuilder-dotnet/pull/3730

**Comment by vipeketi** [comment]
PR is merged. Fixed with #3730
